book-store-ingress.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata: 
+  name: book-store-ingress
+  namespace: book-store-ns
```

Comment:
Why call the namespace `book-store-ns` ?
First, you don't need to add the `ns` (namespace) to the namespace name, it's obvious it's a name space, because it the namespace.
Second, are you sure it's create the `namespace` if it's not already exist? Or should you add a `namespace` yaml configuration before all of that to create it (in case it's not exist already)?
Third, the app is a `book-store` but this is just the backend (let's assume it's production and the cluster is called `prod`), I would have called it something like `backend` or `server` (if it was more specific service like `users` or `auth` I would have called it that, or give it an internal name in the company..)

Reply:
I get what you mean .. I put the `ns` in the just for the good order.. I can remove it.

I have a debt repleted to naming of the `ns`, all the components  of the app should be on the same `ns`, so we shouldn't call the `ns` `backend` or what they serve, we can call it prod. should you meant to call the `deployment` and `service` like that??
 
second, you right about the creation of the namespace, we should first create the `ns` and afterward apply the deployment, unless we will get an err msg that we do not have such `ns`. I created another yaml file for the `ns`.